contracts/Oracle.sol

```diff
@@ -3,10 +3,133 @@
  */
 pragma solidity 0.6.10;
 
+import "./interfaces/AggregatorInterface.sol";
+import "./interfaces/AddressBookInterface.sol";
+import "./packages/oz/Ownable.sol";
+import "./packages/oz/SafeMath.sol";
+
 /**
- *
+ * @author Opyn Team
+ * @title Oracle Module
+ * @notice The Oracle module provide the system with on-chain prices
  */
-// solhint-disable-next-line no-empty-blocks
-contract Oracle {
+contract Oracle is Ownable {
+    using SafeMath for uint256;
+
+    /// @dev structure that represent price, and timestamp
+    struct Price {
+        uint256 price;
+        uint256 timestamp; // timestamp at which the price is pulled to this oracle
+    }
+
+    /// @dev mapping between oracle and it's locking period
+    mapping(address => uint256) internal oracleLockingPeriod;
+    /// @dev mapping of asset price to it's dispute period
+    mapping(address => uint256) internal oracleDisputePeriod;
+    /// @dev mapping between batch and it's oracle
+    mapping(bytes32 => address) internal batchOracle;
+    /// @dev mapping between batch and it price at specific timestmap
+    mapping(bytes32 => mapping(uint256 => Price)) internal batchPriceAt;
+
+    /**
+     * @notice get batch price
+     * @param _batch batch hash
+     * @param _timestamp price timestamp
+     * @return price and timestap at which price submitted to this contract
+     */
+    function getBatchPrice(bytes32 _batch, uint256 _timestamp) public view returns (uint256, uint256) {
+        Price memory batchPrice = batchPriceAt[_batch][_timestamp];
+        return (batchPrice.price, batchPrice.timestamp);
+    }
```

Comment:
Remove this function if we don't have setter and tests in this PR.

Reply:
This one was added in the getters PR, I think we can keep it as I will add tests for it in the next PR.